### PR TITLE
edk2: fix hash

### PR DESCRIPTION
**Pull request by superherointj**
edk2: fix hash

Fixes regression:
- https://github.com/NixOS/nixpkgs/pull/321143#issuecomment-2212458679
- https://github.com/NixOS/nixpkgs/pull/325059#issuecomment-2212454476

CC @adamcstephens @RaitoBezarius

### CODE REVIEW

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 325267` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>quickemu</li>
    <li>quickgui</li>
  </ul>
</details>
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>OVMF</li>
    <li>OVMF-cloud-hypervisor</li>
    <li>OVMF-cloud-hypervisor.fd</li>
    <li>OVMF.fd</li>
    <li>OVMFFull</li>
    <li>OVMFFull.fd</li>
    <li>edk2</li>
    <li>edk2-uefi-shell</li>
    <li>lxd-lts</li>
    <li>multipass</li>
  </ul>
</details>

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 325267` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>quickemu</li>
    <li>quickgui</li>
  </ul>
</details>
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>OVMF</li>
    <li>OVMF-cloud-hypervisor</li>
    <li>OVMF-cloud-hypervisor.fd</li>
    <li>OVMF.fd</li>
    <li>OVMFFull</li>
    <li>OVMFFull.fd</li>
    <li>edk2</li>
    <li>edk2-uefi-shell</li>
    <li>lxd-lts</li>
    <li>multipass</li>
  </ul>
</details>

File: pkgs/by-name/ed/edk2/package.nix [hunk 1]
```diff
@@ -51,7 +51,9 @@ edk2 = stdenv.mkDerivation {
     repo = "edk2";
     rev = "edk2-stable${edk2.version}";
     fetchSubmodules = true;
-    hash = "sha256-7vNodHocwqQiO0ZXtqo8lEOFyt8JkFHcAathEhrKWE0=";
+    # Note: Hash changes whenever OpenSSL package is upgraded,
+    #       whenever upgrading edk2 check for concurrent OpenSSL upgrades
+    hash = "sha256-4JDeA4QBa/FYqgHi5l6FT9EdcTtFvIbKzO+ywT1BsC8=";
```
Comment (Mic92) [inline pkgs/by-name/ed/edk2/package.nix:56]: uff. That's bad. We should create a second derivation that takes this fixed-input derivation as an input and puts a new patched source tree.

Comment (Mic92) [reply pkgs/by-name/ed/edk2/package.nix:56]: Going to push an improved version.

**Comment by Mic92** [comment]
Sorry for the reformat commit but the original formatting was not very readable.

File: pkgs/by-name/ed/edk2/package.nix (hunk 1 above)
Comment (Mic92) [reply pkgs/by-name/ed/edk2/package.nix]: Fixed.

**Review by adamcstephens — APPROVED** [review]

**Comment by RaitoBezarius** [comment]
So is OVMF using the debundled OpenSSL?

**Comment by adamcstephens** [comment]
> So is OVMF using the debundled OpenSSL?

Yes, this appears to be a revert of the non-upgrade changes from my PR, so removal of bundled openssl is a postPatch again

**Comment by vcunat** [comment]
Has anyone tried `nixosTests.boot.uefiUsb`?  (I might have some issues here)
@ofBorg build nixosTests.boot.uefiUsb

**Review by mjoerg — COMMENTED** [review]
I think this still uses the bundled OpenSSL version in `edk2.mkDerivation`.  I gave it a try, and added a custom `updateScript`: https://github.com/mjoerg/nixpkgs/commit/edk2-fix-hash/

**Comment by superherointj** [comment]
> Has anyone tried `nixosTests.boot.uefiUsb`? (I might have some issues here) @ofborg build nixosTests.boot.uefiUsb

At `master`:

image

At `superherointj:edk2-fix-hash`:

image

`timeout reached; test terminating...`

**Review by adamcstephens — CHANGES_REQUESTED** [review]
Changing review as this does not properly devendor OpenSSL for edk2.mkDerivation which is a requirement.

**Comment by adamcstephens** [comment]
There was a bit too much going on here, so I opened https://github.com/NixOS/nixpkgs/pull/325403 which is more of a direct revert back to the previous state.

**Pull request closed without merging**